src/install_rules.ml

```diff
@@ -312,84 +448,16 @@ let promote_install_file (ctx : Context.t) =
+  type t = Super_context.t * Package.t
 
-  let hash (x, y) = Hashtbl.hash (Super_context.hash x, Local_package.hash y)
+  let hash (x, y) = Hashtbl.hash (Super_context.hash x, Package.hash y)
   let equal (x1, y1) (x2, y2) = (x1 == x2 && y1 == y2)
   let to_dyn _ = Dyn.Opaque
 end
 
-let install_entries sctx package =
-  let installs =
-    Local_package.installs package
-    |> List.concat_map ~f:(fun (d : _ Dir_with_dune.t) ->
-      let { Dune_file.Install_conf. section; files; package = _ } =
-        d.data
-      in
-      List.map files ~f:(fun fb ->
-        let loc = File_binding.Expanded.src_loc fb in
-        let src = File_binding.Expanded.src fb in
-        let dst = File_binding.Expanded.dst fb in
-        ( Some loc
-        , Install.Entry.make section src ?dst
-        )))
-  in
-  let docs =
-    (Local_package.mlds package
-     |> List.map ~f:(fun mld ->
-       (None,
-        Install.Entry.make
-          ~dst:(sprintf "odoc-pages/%s" (Path.Build.basename mld))
-          Install.Section.Doc mld)))
-    @
-    (Local_package.odig_files package
-     |> List.map ~f:(fun doc -> (None, Install.Entry.make Doc doc)))
-  in
-  let lib_install_files =
-    Local_package.lib_stanzas package
-    |> List.concat_map
-         ~f:(fun { Dir_with_dune.
-                   data = (lib : Dune_file.Library.t)
-                 ; scope
-                 ; ctx_dir = dir
-                 ; src_dir = _
-                 ; kind = dir_kind
-                 ; dune_version = _
-                 } ->
-              let sub_dir = (Option.value_exn lib.public).sub_dir in
-              let dir_contents = Dir_contents.get sctx ~dir in
-              lib_install_files sctx ~dir ~sub_dir lib ~scope
-                ~dir_kind ~dir_contents)
-  in
-  let coqlib_install_files =
-    Local_package.coqlibs package
-    |> List.concat_map
-         ~f:(fun { Dir_with_dune.
-                   data = coqlib
-                 ; ctx_dir = dir
-                 ; _
-                 } ->
-              Coq_rules.install_rules ~sctx ~dir coqlib)
-  in
-  let metadata =
-    let meta = Local_package.meta_file package in
-    let dune_package = Local_package.dune_package_file package in
-    let opam_file = Local_package.opam_file package in
-    (None, Install.Entry.make Lib meta ~dst:"META")
-    :: (None, Install.Entry.make Lib dune_package
-                ~dst:"dune-package")
-    :: (let package = Local_package.package package in
-        match package.kind with
-        | Dune false -> []
-        | Dune true
-        | Opam ->
-          [(None, Install.Entry.make Lib opam_file ~dst:"opam")])
-  in
-  coqlib_install_files
-  |> List.rev_append lib_install_files
-  |> List.rev_append installs
-  |> List.rev_append docs
-  |> List.rev_append metadata
+let install_entries sctx (package : Package.t) =
+  let packages = Stanzas_to_entries.stanzas_to_entries sctx in
+  Package.Name.Map.Multi.find packages package.name
 
 let install_entries =
```

Comment:
We can remove this. `stanzas_to_entries` is already memoised, so this is caching a `Map.find`, which is likely to be less costly than the overhead of the memoisation system.

Reply:
I ended up doing this and the next change as well